jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -643,6 +643,13 @@ protected void doStart() throws Exception
         if (getContextPath() == null)
             throw new IllegalStateException("Null contextPath");
 
+        Resource baseResource = getBaseResource();
+        if (baseResource != null)
+        {
+            if (!Resources.isReadable(baseResource))
```

Comment:
Should we also test for it being a directory?

Reply:
We didn't before.
I think it was because with URL based resources we cannot always tell if the resource is a directory or not.